apps/client/src/app/core/data-reporting/exploration-result.reporter.ts

```diff
@@ -1,5 +1,62 @@
 import { ExplorationType } from '../../model/other/exploration-type';
+import { DataReporter } from './data-reporter';
+import { Observable } from 'rxjs/Observable';
 
-export interface ExplorationResultReporter {
-  getExplorationType(): ExplorationType;
+export abstract class ExplorationResultReporter implements DataReporter{
+  abstract getDataReports(packets$: Observable<any>): Observable<any[]>;
+  abstract getExplorationType(): ExplorationType;
+
+  createReportsList(stats, resultLog) {
```

Comment:
Could be useful to type the output of this method for better maintainability 😄

Reply:
Sure, where should the model for it be? In the same folder?

Reply:
Yeah, same folder is perfect.